idaes/models/properties/modular_properties/examples/tests/test_BT_PR_SmoothVLE2.py

```diff
@@ -0,0 +1,891 @@
+            )
+            == 0.9763608
+        )
+        assert (
+            pytest.approx(
+                value(m.fs.state[1].fug_coeff_phase_comp["Vap", "toluene"]), 1e-5
+            )
+            == 0.9663611
+        )
+
+        assert (
+            pytest.approx(
+                value(m.fs.state[1].mole_frac_phase_comp["Liq", "benzene"]), 1e-5
+            )
+            == 0.17342
+        )
+        assert (
+            pytest.approx(
+                value(m.fs.state[1].mole_frac_phase_comp["Liq", "toluene"]), 1e-5
+            )
+            == 0.82658
+        )
+        assert (
+            pytest.approx(
+                value(m.fs.state[1].mole_frac_phase_comp["Vap", "benzene"]), 1e-5
+            )
+            == 0.3267155
+        )
+        assert (
+            pytest.approx(
+                value(m.fs.state[1].mole_frac_phase_comp["Vap", "toluene"]), 1e-5
+            )
+            == 0.6732845
+        )
+
+        assert (
+            pytest.approx(value(m.fs.state[1].enth_mol_phase["Liq"]), 1e-5) == 31535.8
+        )
+        assert (
+            pytest.approx(value(m.fs.state[1].enth_mol_phase["Vap"]), 1e-5) == 69175.3
+        )
+        assert (
+            pytest.approx(value(m.fs.state[1].entr_mol_phase["Liq"]), 1e-5) == -369.033
+        )
+        assert (
+            pytest.approx(value(m.fs.state[1].entr_mol_phase["Vap"]), 1e-5) == -273.513
+        )
+
+    @pytest.mark.unit
+    def test_basic_scaling(self, m):
+        assert len(m.fs.state[1].scaling_factor) == 23
+        assert m.fs.state[1].scaling_factor[m.fs.state[1].flow_mol] == 1e-2
+        assert m.fs.state[1].scaling_factor[m.fs.state[1].flow_mol_phase["Liq"]] == 1e-2
+        assert m.fs.state[1].scaling_factor[m.fs.state[1].flow_mol_phase["Vap"]] == 1e-2
+        assert (
+            m.fs.state[1].scaling_factor[
+                m.fs.state[1].flow_mol_phase_comp["Liq", "benzene"]
+            ]
+            == 1e-2
+        )
+        assert (
+            m.fs.state[1].scaling_factor[
+                m.fs.state[1].flow_mol_phase_comp["Liq", "toluene"]
+            ]
+            == 1e-2
+        )
+        assert (
+            m.fs.state[1].scaling_factor[
+                m.fs.state[1].flow_mol_phase_comp["Vap", "benzene"]
+            ]
+            == 1e-2
+        )
+        assert (
+            m.fs.state[1].scaling_factor[
+                m.fs.state[1].flow_mol_phase_comp["Vap", "toluene"]
+            ]
+            == 1e-2
+        )
+        assert (
+            m.fs.state[1].scaling_factor[m.fs.state[1].mole_frac_comp["benzene"]]
+            == 1000
+        )
+        assert (
+            m.fs.state[1].scaling_factor[m.fs.state[1].mole_frac_comp["toluene"]]
+            == 1000
+        )
+        assert (
+            m.fs.state[1].scaling_factor[
+                m.fs.state[1].mole_frac_phase_comp["Liq", "benzene"]
+            ]
+            == 1000
+        )
```

Comment:
For the record, I don't think 1000 is a great value for scaling factors of mole fractions. Testing mole fractions like this makes it a pain to change later. Can't we do something like test  Jacobian rows, columns, and condition number?

Reply:
Agreed, but for now I am focused on getting this working (plus the new tools will supersede this before too long).